### ISSUE TITLE
Use vim-table-mode for vim-orgmode table feature.

### ISSUE THREAD

**Issue by dkinzer** [jceb/vim-orgmode#159]
[dhruvasagar/vim-table-mode](https://github.com/dhruvasagar/vim-table-mode) is a feature rich ascii table manipulator that seems to only be lacking a sorting feature and an export feature.  At a glance, the code base seems pretty solid (though, it's vimscript not python); but the test suite looks very comprehensive.

I think incorporating this module into vim-orgmode could be very beneficial.

**Comment by Shurakai** [comment]
@dkinzer @sotte What exactly is left here to do?

I just tried to use it and it seems to work quite nicely; I'm a bit confused though, as in the documentation (section "Creating table on-the-fly") each row seems to be separated by some
filling row.

This does not seem to be the case for org-mode (see http://orgmode.org/manual/Built_002din-table-editor.html#Built_002din-table-editor) and we might have to set the option accordingly, see the

   call s:SetGlobalOptDefault('table_mode_fillchar', '-')
  
at https://github.com/dhruvasagar/vim-table-mode/blob/master/plugin/table-mode.vim

**Comment by sotte** [comment]
@Shurakai the table feature of emacs' orgmode is really a beast. It can do so many things: `Tab` to jump to the next cell, auto format on every `Tab`, do math on the cells (replace Excel), fill cells with previously defined data, etc.

So I guess there is a lot left to do :)

But if "just" want to integrate vim-table-mode, we should create a pull request and talk about the details.

**Comment by Shurakai** [comment]
Yeah I think we should just go ahead with a PR. There are many other parts of vim-orgmode that require attention and vim-table-mode is quite sophisticated.

We need to add documentation for sure, but what else? It worked quite well already.

**Comment by sotte** [comment]
Documentation, tests...
And the plugin should be optional, i.e., nothing breaks if vim-table-mode is not installed.

**Comment by jalcine** [comment]
Curious to know if a PR's been crafted; I'm willing to test drive against a few cases of live data (or write tests for said examples).

**Comment by doak** [comment]
@dkinzer, why was this issue closed? Although no code is there, the comments here seem to be a good blueprint to put things together. And closed means either "not valid" or "done", imho neither is correct.

@Shurakai, did you managed to implement something?

**Comment by 0xACE** [comment]
![159](https://user-images.githubusercontent.com/1482942/88961913-35098d80-d295-11ea-8b16-aa44a0505e8c.gif)

Not sure about MIT vs GPL licensing and what functionality you want... but vimwikis implementation automatically aligns the table whenever you exit `insertmode` or tab to the next cell...

Let me know and I'll submit my PR mid september.

**Comment by farao** [comment]
> Let me know and I'll submit my PR mid september.

This would be very helpful for me! Since I did not see a merged or open PR about this: can I find the code somewhere or would you provide this as a PR @0xACE ? :-)
